### PR TITLE
Add some information on reduce! :D

### PR DESCRIPTION
**Pull request by DavidAwad**
It's a super useful function and always gets taught with map and filter

### CODE REVIEW

File: map_filter.rst [hunk 1]
```diff
@@ -75,3 +75,29 @@ The filter resembles a for loop but it is a builtin function and faster.
...
+the product of a list of integers. 
+
+So the normal way you might go about doing this task in python us using
+a basic for loop.
+
+.. code:: python
+    product = 1
+    list = [1, 2, 3, 4]
+    for num in list: 
+        product = product * list
```
Comment (yasoob) [inline map_filter.rst]: It should be `product = product * num`

**Comment by DavidAwad** [comment]
doh

**Comment by DavidAwad** [comment]
yeah I'll fix that @yasoob

**Comment by DavidAwad** [comment]
@yasoob  stupid typos fixed

**Comment by DavidAwad** [comment]
and PR corrected

**Pull request merged**